src/python/ensembl/io/genomio/genome_metadata/dump.py

```diff
@@ -130,7 +131,7 @@ def check_assembly_version(gmeta_out: Dict[str, Any]) -> None:
     version = assembly.get("version")
 
     # Check the version is an integer
-    if version is not None and version.isdigit():
+    if version is not None and (isinstance(version, int) or version.isdigit()):
```

Comment:
This code can still raise an exception if `version` is a dictionary (although unlikely). This, it may be best to do:
```
try:
    assembly["version"] = int(version)
except (ValueError, TypeError):
    # Get the version from the assembly accession
    ...
else:
    logging.info(f"Located version [v{int(version)}] info from meta data.")
```
_Note:_ unfortunately I cannot suggest this change for lines untouched by this PR, so you will need to apply the suggestion manually. Sorry!